### PR TITLE
add \cancel function.

### PR DESCRIPTION
**Pull request by AlabasterAxe**
## Description

This PR adds a single operand function \cancel which draws a slash from the bottom left corner to the top right corner of the child as described by http://tug.ctan.org/tex-archive/macros/latex/contrib/cancel/cancel.pdf

## Checklist

*Remove `If [...]` items that do not apply to your PR.*

- [x] I have made myself familiar with the CaTeX
      [contributing guide](https://github.com/simpleclub/CaTeX/blob/master/CONTRIBUTING.md).
- [X] I added a PR description.
- [X] If this PR changes anything about the main `catex` or `example` package (also README etc.),
      I created an entry in `CHANGELOG.md` (`## UPCOMING RELEASE` if the change on its own
      is not worth an update).
- [X] If this PR adds new functions, I updated the `function_prioritization.csv` file.
- [x] If there is new functionality in code, I added tests covering all my additions
      (Golden tests for anything rendering related). - I didn't see any place to add golden tests.
- [X] All required checks pass.

### CODE REVIEW

**Comment by AlabasterAxe** [comment]
Some examples of it in action:
cancel

File: lib/src/rendering/functions/cancel.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  @override
+  void configure() {
+    final childSize = sizeChildNode(children[0]);
+
+    renderSize = childSize;
+
+    _linePaint = Paint()
+      ..strokeWidth = 2
+      ..color = context.color;
```
Comment (creativecreatorormaybenot) [inline lib/src/rendering/functions/cancel.dart:21]: This should probably be a local field in the `render` method.

File: lib/src/rendering/functions/cancel.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  @override
+  void render(Canvas canvas) {
+    paintChildNode(children[0]);
+    final double horizontalPadding = renderSize.width * .05;
+    final double verticalPadding = renderSize.height * .05;
```
Comment (creativecreatorormaybenot) [inline lib/src/rendering/functions/cancel.dart:28]: We use implicit types.

```suggestion
    final horizontalPadding = renderSize.width * .05;
    final verticalPadding = renderSize.height * .05;
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## UPCOMING RELEASE
+
+* Add `\cancel` function.
+
```
Comment (creativecreatorormaybenot) [inline CHANGELOG.md:4]: You can do `0.0.1+8` if you like to :)

You should then also adjust it in `pubspec.yaml`.

**Review by creativecreatorormaybenot — COMMENTED** [review]
Thanks so much for your contribution @AlabasterAxe ❤️

I am not sure if you have seen the on-hold notice - this is why there is no null safety support and why function support is very limited.  
(why this is a pre-0.1.0 version essentially)

Still, your contribution is much appreciated and if the comments are resolved, I think we can merge this 🚀   
Now, I am tempted to migrate this to null safety, just because it seems like someone is using it right now -  
On the flip side, it would require a version bump to `0.1.0` because of the breaking change and this package is not yet ready for `0.1.0`. I think I might migrate it to null safety anyway and just break the package in `0.0.2` 👍

File: lib/src/rendering/functions/cancel.dart (hunk 1 above)
Comment (creativecreatorormaybenot) [reply lib/src/rendering/functions/cancel.dart:21]: This is because it is strictly only relevant to rendering.

**Comment by AlabasterAxe** [comment]
Hi @creativecreatorormaybenot happy to help and thanks for calling out the on-hold status more explicitly! We've really enjoyed using this in our project but my teammate and I have taken a look at flutter_math per your on-hold note's suggestion and have decided we will probably try to move forward with that. I've made your proposed changes since we're already quite close anyway.

Thanks!

**Review by creativecreatorormaybenot — APPROVED** [review]

**Comment by creativecreatorormaybenot** [comment]
@AlabasterAxe We have created a **fork** of `flutter_math` that ensures that it continues to work (@znjameswu has not updated the `flutter_math` repo this year). See https://github.com/simpleclub-extended/flutter_math_fork. We advise you to use that until we pick up the CaTeX project again ✨ 

I will merge your PR, thanks.

**Pull request merged**